modules/developer_manual/pages/core/acceptance-tests.adoc

```diff
@@ -300,5 +221,306 @@ If you want to have encryption enabled add `OC_TEST_ENCRYPTION_ENABLED=1`, as in
+The user name of the administrator needs to be determined at run-time, not hard-coded in the scenario.
+
+==== Referring to Named Entities
+
+When referring to specific named entities on the system, such as a user, group, file, folder or tag,
+then do not put the word `the` in front, but do put the name of the entity. For example:
+
+[source,gherkin]
+----
+    Given user "user0" has been added to group "grp1"
+    And user "user0" has uploaded file "abc.txt" into folder "folder1"
+    And user "user0" has added tag "aTag" to file "folder1/abc.txt"
+    When user "user0" shares folder "folder1" with user "user1"
+    ...
+----
+
+This makes it clearer to understand which entity is required in which position of the sentence.
+For example:
+
+[source,gherkin]
+----
+    And "user0" has uploaded "abc.txt" into "folder1"
+    ...
+----
+
+would be less clear that the required entities for this step are a user, file and folder.
+
+=== Scenario Background
+
+If all the scenarios in a feature start with a common set of `Given` steps,
+then put them into a `Background:` section. For example:
+
+[source,gherkin]
+----
+  Background:
+    Given user "user0" has been created
+    And user "user1" has been created
+    And user "user0" has uploaded file "abc.txt"
+
+  Scenario: share a file with another user
+    When user "user0" shares file "abc.txt" with user "user1" using the sharing API
+    Then the HTTP status code should be "200"
+    And user "user1" should be able to download file "abc.txt"
+
+  Scenario: share a file with a group
+    Given group "grp1" has been created
+    And "user1" has been added to group "grp1"
+    When user "user0" shares file "abc.txt" with user "user1" using the sharing API
+    Then the HTTP status code should be "200"
+    And user "user1" should be able to download file "abc.txt"
+----
+
+This reduces some duplication in feature files.
+
+[[how-to-add-a-new-feature]]
+== How to Add New Test Steps
+
+To do - write this section. The following is some code that was already in this document:
+
+The first thing we need to do is create a new file for the context; we'll name it `TaskToTestContext.php`.
+In the file, we'll add the code snippet below:
+
+[source,php]
+----
+<?php
+
+use Behat\Behat\Context\Context;
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+/**
+ * Example Context.
+ */
+class ExampleContext implements Context {
+  use Webdav;
+}
+----
+
+Here's example code for a scenario:
+
+[source,php]
+----
+/**
+ * @When Sending a :method to :url with requesttoken
+ *
+ * @param string $method
+ * @param string $url
+ */
+public function exampleFunction($method, $url) {
+----
+
+== References
+
+For more information on Behat, and how to write acceptance tests using it, see http://behat.org/en/latest/guides.html[the Behat documentation].
+
+For background information on BDD, see https://dannorth.net/whats-in-a-story/[Dan North resources].
```

Comment:
Mind to write down what BDD is, else this is just an acronym hard to understand:
Behaviour-driven development (BDD)

Reply:
Done